modules/gdscript/gdscript_parser.cpp

```diff
@@ -3952,28 +3952,21 @@ GDScriptParser::DataType GDScriptParser::SuiteNode::Local::get_datatype() const
 }
 
 String GDScriptParser::SuiteNode::Local::get_name() const {
-	String name;
 	switch (type) {
 		case SuiteNode::Local::PARAMETER:
-			name = "parameter";
-			break;
+			return "parameter";
 		case SuiteNode::Local::CONSTANT:
-			name = "constant";
-			break;
+			return "constant";
 		case SuiteNode::Local::VARIABLE:
-			name = "variable";
-			break;
+			return "variable";
 		case SuiteNode::Local::FOR_VARIABLE:
-			name = "for loop iterator";
-			break;
+			return "for loop iterator";
 		case SuiteNode::Local::PATTERN_BIND:
-			name = "pattern_bind";
-			break;
+			return "pattern_bind";
 		case SuiteNode::Local::UNDEFINED:
-			name = "<undefined>";
-			break;
+			return "<undefined>";
 	}
-	return name;
+	return String();
```

Comment:
This may raise a warning for unreachable code since the above switch handles all cases.

Reply:
And without it, GCC complains about "non-void" function without return.

Reply:
Haha yes I hate C++ enums which don't guarantee that the value is contained in the enum.

The better option might be to `return String()` in a `default:` case.

Reply:
But if this doesn't raise a new warning, it's fine as is of course.